public/js/main.js

```diff
@@ -222,66 +326,19 @@ $(document).ready(() => {
   });
   $('#addIndicatorBtn_SMA').click(() => {
     console.log('### Inside addIndicator_SMA');
-    const SMAParam = $('#addIndicator_SMA').val();
 
-    // Parsing pattern for csv SMA values
-    const pattern = /\s*,\s*/;
-    const userInputChartList = SMAParam.split(pattern);
-
-    // Transferring list of current charts to seperate array, ignoring i = 0
-    //  and i = mainChart.series.length - 1, as those are occupied by the
-    //  candlestick data and the series info display respectively
-    const chartList = [];
-    for (let i = 1; i !== mainChart.series.length - 1; i += 1) {
-      chartList.push(mainChart.series[i].name);
-    }
-
-    // This set of functions manages multiple entries of csv inputs, and updates
-    //  the chart to the newest input. For example, if 5, 10 is entered first,
-    //  5 and 10 indicators are added. If 10, 15 is entered next, 5 is removed,
-    //  10 remains, and 15 is added.
-
-    // Removing any series not specified in userInputChartList (the array of
-    //  indicators to be displayed)
-    for (let i = 0; i !== chartList.length; i += 1) {
-      let shouldExist = false;
-      for (let j = 0; j !== userInputChartList.length; j += 1) {
-        let tmpUICL = "SMA - " + userInputChartList[j];
-        if (chartList[i] === tmpUICL) {
-          shouldExist = true;
-        }
-      }
-      // If a currently existing series does not appear anywhere in the user
-      //  inputted list of indicators, remove it
-      if (!shouldExist) {
-        mainChart.get(chartList[i]).remove();
-      }
-    }
+    const SMAParam = $('#addIndicator_SMA').val();
+    csvToAddData(chartData, mainChart, SMAParam, 'SMA');
 
-    // Adding new series from the user inputted list
-    for (let i = 0; i !== userInputChartList.length; i += 1) {
-      let exists = false;
-      for (let j = 0; j !== chartList.length; j += 1) {
-        let tmpUICL = "SMA - " + userInputChartList[i];
-        if (chartList[j] === tmpUICL) {
-          exists = true;
-        }
-      }
-      // If a user inputted indicator appears nowhere in the the current chart
-      //  add it
-      if (!exists) {
-        let SMA = window.SMA(chartData, userInputChartList[i]);
-        // Creating a unique id for this indicator
-        let id = 'SMA - ' + userInputChartList[i];
+    // Redrawing chart after adding indicators
+    mainChart.redraw();
+  });
+  $('#addIndicatorBtn_EMA').click(() => {
```

Comment:
Try to avoid IDs when possible! DOM query times have gotten fast enough that there's not an imperative to use them from a performance aspect, and you can define much more semantically-informative selectors using data attributes (i.e. `$('[data-click=add-indicator]').click( ... )`).

This also comes with the added benefit of fully decoupling your functional markup (i.e. markup to which event handlers are assigned) from your stylistic markup (i.e. classes and ids).